lib/ansible/plugins/strategy/__init__.py

```diff
@@ -1114,15 +917,17 @@ def _evaluate_conditional(h):
         self._tqm.send_callback('v2_playbook_on_task_start', task, is_conditional=False)
 
         # These don't support "when" conditionals
-        if meta_action in ('noop', 'flush_handlers', 'refresh_inventory', 'reset_connection') and task.when:
+        if meta_action in ('noop', 'refresh_inventory', 'reset_connection') and task.when:
             self._cond_not_supported_warn(meta_action)
 
         if meta_action == 'noop':
             msg = "noop"
         elif meta_action == 'flush_handlers':
-            self._flushed_hosts[target_host] = True
-            self.run_handlers(iterator, play_context)
-            self._flushed_hosts[target_host] = False
+            host_state = iterator.get_state_for_host(target_host.name)
+            # prevent flush_handlers in a handler
+            if target_host.name not in self._tqm._unreachable_hosts and host_state.run_state not in (IteratingStates.HANDLERS, IteratingStates.COMPLETE):
```

Comment:
Maybe leniency is better if handlers eventually support full blocks with rescue/always, but I sort of feel like it should just be an error if someone puts a `flush_handlers` in a handler.

Reply:
I think you are right, this shouldn't be silently ignored.

(`leniency` - I learned a new word ;-) )

Reply:
Done.